pkg/ibmcsidriver/node_helper_test.go

```diff
@@ -20,18 +20,87 @@
 package ibmcsidriver
 
 import (
+	"fmt"
 	"testing"
 
 	cloudProvider "github.com/IBM/ibmcloud-volume-file-vpc/pkg/ibmcloudprovider"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestProcessMount(t *testing.T) {
+	// test cases
+	testCases := []struct {
+		name              string
+		stagingTargetPath string
+		targetPath        string
+		fsType            string
+		options           []string
+		expectedErr       error
+	}{
+		{
+			name:              "success",
+			stagingTargetPath: "/staging",
+			targetPath:        "/targetpath",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       nil,
+		},
+		{
+			name:              "Invalid staging path",
+			stagingTargetPath: "./error_mount_source",
+			targetPath:        "/targetpath",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       fmt.Errorf("fake Mount: source error"),
+		},
+		{
+			name:              "Invalid target path",
+			stagingTargetPath: "./error_mount_target",
+			targetPath:        "fake-volPath",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       fmt.Errorf("fake Mount: target error"),
+		},
+		{
+			name:              "Make directory fails",
+			stagingTargetPath: "/staging",
+			targetPath:        "invalid-volPath-dir",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       fmt.Errorf("Path Creation failed"),
+		},
+		{
+			name:              "IsLikelyNotMountPoint returns true and nil error",
+			stagingTargetPath: "./error_mount_source",
+			targetPath:        "fake-volPath-1",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       fmt.Errorf("fake Mount: target error"),
+		},
+		{
+			name:              "Umount Fails",
+			stagingTargetPath: "./error_mount_source",
+			targetPath:        "error_umount",
+			fsType:            "ext4",
+			options:           []string{"ro,sync"},
+			expectedErr:       fmt.Errorf("Unmount Failed"),
+		},
+	}
+
 	// Creating test logger
 	logger, teardown := cloudProvider.GetTestLogger(t)
 	defer teardown()
 
-	icDriver := initIBMCSIDriver(t)
-	ops := []string{"a", "b"}
-	response, err := icDriver.ns.processMount(logger, "processMount", "/staging", "/targetpath", "ext4", ops)
-	t.Logf("Response %v, error %v", response, err)
+	// Run test cases
+	for _, tc := range testCases {
+		t.Logf("test case: %s", tc.name)
+		// Setup new driver each time so no interference
+		icDriver := initIBMCSIDriver(t)
+		// Call processMound
+		_, err := icDriver.ns.processMount(logger, "processMount", tc.stagingTargetPath, tc.targetPath, tc.fsType, tc.options)
```

Comment:
same here,
are we really need staging, target and fs type all 3 parameter in file csi driver? looks not needed please check and update the PR accordingly

Reply:
@arahamad changed the staging variable to source to avoid confusion. Also fs type is mandatory parameter for mount command. Please check once